app/models/pageflow/membership.rb

```diff
@@ -1,9 +1,26 @@
 module Pageflow
   class Membership < ActiveRecord::Base
     belongs_to :user
-    belongs_to :entry
+    belongs_to :entity, polymorphic: true
 
-    validates :user, :entry, :presence => true
-    validates :entry_id, :uniqueness => { :scope => :user_id }
+    validates :user, :entity, :role, presence: true
+    validates :user_id, uniqueness: {scope: [:entity_type, :entity_id]}
+    if :entity_type == 'Pageflow::Entry'
+      validates_with AccountMembershipExistenceValidator
+      validates :role, inclusion: {in: %w(previewer editor publisher manager).map(&:to_sym)}
+    else
+      validates :role, inclusion: {in: %w(member previewer editor publisher manager).map(&:to_sym)}
```

Comment:
Again I am wondering if our lifes were simpler if roles always were strings. Specs are the only cases were they can be symbols anyway, right? Coming from the request or the db, they will be strings.